picard/ui/metadatabox/tagdiff.py

```diff
@@ -342,3 +345,28 @@ def update_tag_names(self, changes_first=False, top_tags=None):
             self.tag_names = [
                 tag for tag in tag_names if
                 self.status[tag] != TagStatus.EMPTY]
+
+    def to_json(self):
+        result = {}
+        for tag in self.tag_names:
+            result[tag] = {}
+            if tag in self.old:
+                result[tag]['old'] = self.old[tag]
+            if tag in self.new:
+                result[tag]['new'] = self.new[tag]
+
+        return json.dumps(result)
+
+    def to_tsv(self):
+        COL_SEP = "\t"
+
+        f = io.StringIO()
+        writer = csv.writer(f, delimiter=COL_SEP)
```

Comment:
Then you should use a dialect to make things clear, see `excel-tab` dialect (https://docs.python.org/3/library/csv.html#csv.excel_tab).


```python
import io
import csv

csvdata = ["1", "2\t3", "4\n5", '6"7', '']

dialect = 'excel-tab'
target = io.StringIO()
writer = csv.writer(target, dialect=dialect)
writer.writerow(csvdata)
csvresult = target.getvalue()
print(csvresult)

source = io.StringIO(csvresult)
reader = csv.reader(source, dialect=dialect)
for row in reader:
    print(row)
    print(row == csvdata)

```

@phw: do we need to handle encoding somehow?